### PR TITLE
Bumped up PythonConnector version from 3.7.0 to 3.7.1

### PR DESCRIPTION
**Pull request by sfc-gh-smirzaei**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Preparation for v3.7.1 release.

### CODE REVIEW

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-aling — APPROVED** [review]

**Comment by sfc-gh-smirzaei** [comment]
I have read the CLA Document and I hereby sign the CLA

File: DESCRIPTION.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
-- v3.7.1(TBD)
+- v3.7.1(February 21, 2024)
 
   - Bumped pandas dependency from >=1.0.0,<2.2.0 to >=1.0.0,<3.0.0.
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:13]: Bumped the pandas.... What do you think of using something like from 1.0.0-2.2.0 to 1.0.0-3.00?

(same for next one)

**Review by sfc-gh-mheavin — COMMENTED** [review]
Feel free to slack me if you have questions.

**Review by sfc-gh-mheavin — APPROVED** [review]
Approved based on your decision

**Comment by sfc-gh-smirzaei** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**